test/framework/applications/guestbooktest.go

```diff
@@ -54,55 +52,76 @@ const (
-	if _, err := os.Stat(templateFilepath); err != nil {
-		return nil, fmt.Errorf("could not find Guest book template in '%s'", templateFilepath)
-	}
-
-	guestBooktpl := template.Must(template.ParseFiles(templateFilepath))
 	return &GuestBookTest{
-		resourcesDir: resourcesDir,
-		chartRepo:    filepath.Join(resourcesDir, "charts"),
-		guestBookTpl: guestBooktpl,
+		framework:        f,
+		guestBookAppHost: fmt.Sprintf("guestbook-%s.ingress.%s", randURLSuffix, *f.Shoot.Spec.DNS.Domain),
 	}, nil
 }
 
 // WaitUntilPrerequisitesAreReady waits until the redis master is ready.
-func (t *GuestBookTest) WaitUntilPrerequisitesAreReady(ctx context.Context, gardenerTestOperations *framework.GardenerTestOperation) {
-	err := gardenerTestOperations.WaitUntilStatefulSetIsRunning(ctx, RedisMaster, helmDeployNamespace, gardenerTestOperations.ShootClient)
+func (t *GuestBookTest) WaitUntilPrerequisitesAreReady(ctx context.Context) {
+	err := t.framework.WaitUntilStatefulSetIsRunning(ctx, RedisMaster, t.framework.Namespace, t.framework.ShootClient)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 }
 
+// WaitUntilGuestbookAppIsAvailable waits until the deployed guestbook application can be reached via http
+func (t *GuestBookTest) WaitUntilGuestbookAppIsAvailable(ctx context.Context, guestbookAppUrls []string) error {
+	defaultPollInterval := time.Minute
+	return retry.UntilTimeout(ctx, defaultPollInterval, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
+		for _, guestbookAppURL := range guestbookAppUrls {
+			response, err := framework.HTTPGet(ctx, guestbookAppURL)
+			if err != nil {
+				t.framework.Logger.Infof("Guestbook app url: %q is not available yet: %s", guestbookAppURL, err.Error())
+				return retry.MinorError(err)
+			}
+
+			if response.StatusCode != http.StatusOK {
+				t.framework.Logger.Infof("Guestbook app url: %q is not available yet", guestbookAppURL)
+				return retry.MinorError(fmt.Errorf("guestbook app url %q returned status %s", guestbookAppURL, response.Status))
+			}
+
+			responseBytes, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return retry.SevereError(err)
+			}
+
+			bodyString := string(responseBytes)
+			if strings.Contains(bodyString, "404") || strings.Contains(bodyString, "503") {
+				t.framework.Logger.Infof("Guestbook app is not ready yet")
+				return retry.MinorError(fmt.Errorf("guestbook response body contained an error code"))
+			}
+		}
+		t.framework.Logger.Infof("Rejoice, the guestbook app urls are available now!")
+		return retry.Ok()
+	})
+}
+
 // DeployGuestBookApp deploys the redis helmchart and the guestbook app application
-func (t *GuestBookTest) DeployGuestBookApp(ctx context.Context, gardenerTestOperations *framework.GardenerTestOperation) {
-	shoot := gardenerTestOperations.Shoot
+func (t *GuestBookTest) DeployGuestBookApp(ctx context.Context) {
+	if t.framework.Namespace == "" {
+		_, err := t.framework.CreateNewNamespace(ctx)
```

Comment:
I would expect `CreateNewNamespace` to not set `f.framework` as I'm already checking for it here.

Reply:
I need to also set the namespace on shoot.framework as the cleanup is done in the frameworks `AfterEach` cleanup step.

Maybe I should rename the method so that the caller knows that the namespace is set for the framework? `f.CreateAndSetNewNamespace`

Reply:
@mvladev @schrodit  is this resolved?

Reply:
@mvladev ping